pkg/kubernetes/client/get.go

```diff
@@ -39,7 +39,12 @@ func (k Kubectl) GetByLabels(namespace, kind string, labels map[string]string) (
 // resource in the state
 func (k Kubectl) GetByState(data manifest.List) (manifest.List, error) {
 	list, err := k.get("", "", []string{"-f", "-"}, getOpts{
-		stdin: data.String(),
+		// TODO: is this the right place to specify --ignore-not-found?
+		// 	or do we want to pass it as an argument to GetByState()
+		// 	the side effect of adding it here is that prune no longer
+		//	fails when there are new k8s objects in jsonnet
+		ignoreNotFound: true,
```

Comment:
This is certainly tricky. Generally errors should be handed back to the consumer of any given function and handled there, but this case is specific as it implicitly iterates over all elements in state for performance reasons.

I think it would be a good start to have an argument to control that behavior, as `GetByState` could be useful to other consumers that might want strict error handling

Reply:
I have added `GetByStateOpts` to the `client` package. I considered adding `GetOpts` instead but I saw no benefit in allowing consumers to pass `--ignore-not-found` to other calls then `GetByStatus()`. Since there are currently no options to be passed to get operations we only have `GetByStateOpts`.